net: mqtt: Fix typo "seg_tag_list"

mschwan-phytec opened the pull request:
This fixes a typo in `include/net/mqtt.h` where it was "seg_tag_list" instead of "sec_tag_list".

jukkar reviewed (changes requested):
Looks ok, but I think you need to fix also the users of this field
```
$ ack --cc seg_tag_list
subsys/net/lib/mqtt_sock/mqtt_transport_socket_tls.c
58:	if (tls_config->seg_tag_list != NULL && tls_config->sec_tag_count > 0) {
60:				 TLS_SEC_TAG_LIST, tls_config->seg_tag_list,

include/net/mqtt.h
333:	sec_tag_t *seg_tag_list;

samples/net/mqtt_publisher/src/main.c
283:	tls_config->seg_tag_list = m_sec_tags;
```

mschwan-phytec committed:
net: mqtt: Fix typo "seg_tag_list"

Fix a typo in the mqtt_sec_config struct where it was "seg_tag_list"
instead of "sec_tag_list".

Signed-off-by: Martin Schwan <m.schwan@phytec.de>

mschwan-phytec commented:
Seems like I was a bit hasty on this. I changed the typos in the users of the field too and added it to the commit.

rlubos reviewed (approved):
LGTM
